File: src/main/java/net/atos/zac/policy/input/UserInput.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.policy.input;

import jakarta.json.bind.annotation.JsonbProperty;

import net.atos.zac.authentication.LoggedInUser;

public class UserInput {

  @JsonbProperty("user")
  private final UserData userData = new UserData();

  public UserInput(final LoggedInUser loggedInUser) {
    userData.id = loggedInUser.getId();
    userData.rollen = loggedInUser.getRoles();
    userData.zaaktypen =
        loggedInUser.isGeautoriseerdVoorAlleZaaktypen()
            ? null
            : loggedInUser.getGeautoriseerdeZaaktypen();
  }

  public UserData getUser() {
    return userData;
  }
}

File: src/main/java/net/atos/zac/policy/input/UserInput.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.policy.input;

import java.util.Objects;

import jakarta.json.bind.annotation.JsonbProperty;

import nl.info.zac.authentication.LoggedInUser;

public class UserInput {

    @JsonbProperty("user")
    private final UserData userData = new UserData();

    public UserInput(final LoggedInUser loggedInUser) {
        Objects.requireNonNull(loggedInUser,
                "No logged in user found. Please ensure that user did not log out and its session is still active");
        userData.id = loggedInUser.getId();
        userData.rollen = loggedInUser.getRoles();
        userData.zaaktypen = loggedInUser.isAuthorisedForAllZaaktypen() ? null : loggedInUser.getGeautoriseerdeZaaktypen();
    }

    public UserData getUser() {
        return userData;
    }
}
